src/reuse/report.py

```diff
@@ -74,30 +74,79 @@ def __init__(self, do_checksum: bool = True):
         self._files_without_copyright = None
 
     def to_dict(self):
-        """Turn the report into a json-like dictionary."""
-        return {
-            "path": str(Path(self.path).resolve()),
-            "licenses": {
-                identifier: str(path)
-                for identifier, path in self.licenses.items()
-            },
-            "bad_licenses": {
-                lic: [str(file_) for file_ in files]
-                for lic, files in self.bad_licenses.items()
-            },
-            "deprecated_licenses": sorted(self.deprecated_licenses),
-            "licenses_without_extension": {
-                identifier: str(path)
-                for identifier, path in self.licenses_without_extension.items()
+        """Collects and formats data from report and returns it as a dictionary
+
+        :param report: ProjectReport object
+        :return: Dictionary containing data from the ProjectReport object
+        """
+        # Setup report data container
+        data = {
+            "json_version": "1.0",
```

Comment:
Renamed to `lint_version`.

Reply:
Opened issue #728 for the schema.

Reply:
`lint_version` was inspired by `lock-version` in `poetry.lock`: https://github.com/fsfe/reuse-tool/blob/bb3d76fe88298ad3ccc0c88320ef16ffaebb4aea/poetry.lock#L841

The rationale was that there could be other `*_version` fields if we output `--json` for other commands.

I did some research to see if there's a standardised way to do this by referencing the schema directly, like is often done in XML: there isn't. https://github.com/json-schema-org/json-schema-spec/issues/828

We could nevertheless just _do that_. Instead of `lint_version` (or other), use `{"$id": "https://reuse.software/link/to/1.0.0/lint_schema.json", ...}` (in the lint output). I chose `$id` because it most semantically matches the field used _inside of the schema_, see:

- https://json-schema.org/draft/2020-12/json-schema-core.html#section-8.2.1 (still in draft)
- https://json-schema.org/learn/getting-started-step-by-step.html (User-friendly reading)

We could choose a different field, too. `$schema` and `schema` are also candidates.

In order to get all of that right, though, we have to commit to writing a json-schema in order to close this PR. Is that a commitment we want to do, or is that too much effort to get this rolling?

Reply:
I have just confirmed that vscode uses the `$schema` key to reference the schema, which then allows autocompletion by the editor.

Reply:
I'm not mixing issues :) Reiterating what you say to be on the page, the version we're discussing is specific to the output of `reuse lint --json`. If it changes (e.g., field gets added, renamed), that specific version gets bumped, and the package/spec versions change independently as needed.

The reason we may want a schema for the lint JSON output is so that programmers can use that as a sort of documentation for processing the output. The other reason for making a schema is that it would make the versioning of the output a lot easier to reason about; you can just `diff` the two different schema versions.

And in a scenario where we commit to creating schema files, we might as well put a link to the (versioned!) schema file in our output. This has the following benefits:

- We won't need to communicate the schema information out-of-band. `lint_version` refers to the version of a particular schema, but the link to the schema would have to be documented _elsewhere_. It unambiguously answers the question: what is this a version _of_?
- Increased visibility of the schema file. Programmers are lazy and don't read documentation. By putting a link to the schema file in the output itself, they may actually use it to inform their design choices.
- In a hypothetical future, including this key may enable _other_ programs to facilitate interactions with the file. That's what the vscode example was for; not to enable easier editing of the file, but to demonstrate that other programs can automatically detect the declared schema and use it to enhance certain operations. I was trying to detect what the exact field is that other programs are standardising on; it appears to be `$schema`.

Reply:
OK, you convinced me of the advantages of schema files. So we'd include a `$schema` key. But how to version it then? Would the URL to the schema contain a version number? Or would that be a separate key?

In the case of the latter, I once again would prefer to not use the name `lint_version`, but instead `lint_schema_version` or so to clarify that this is not about the REUSE spec version.

Reply:
I don't think it's necessary to group the keys under `versions`. Otherwise I think we're in agreement. The power of consensus :D

Reply:
I'll resolve this here now, as @carmenbianca already opened another issue for that.